src/overlays/gamestates/ovl_title/z_title.c

```diff
@@ -109,62 +109,58 @@ void Title_Draw(GameState* thisx) {
                             1 << 10, 1 << 10);
     }
 
-    func_800FC444(this->gameState.gfxCtx, 0, 0, 0, this->coverAlpha, 2);
+    func_800FC444(this->state.gfxCtx, 0, 0, 0, this->coverAlpha, 2);
 
-    titleRotation += 300;
+    sTitleRotation += 300;
 
-    CLOSE_DISPS(this->gameState.gfxCtx);
+    CLOSE_DISPS(this->state.gfxCtx);
 }
 
-void Title_Main(GameState* thisx) {
-    TitleContext* this = (TitleContext*)thisx;
+void ConsoleLogo_Main(GameState* thisx) {
+    ConsoleLogoState* this = (ConsoleLogoState*)thisx;
 
-    func_8012CF0C(this->gameState.gfxCtx, true, true, 0, 0, 0);
+    func_8012CF0C(this->state.gfxCtx, true, true, 0, 0, 0);
 
-    OPEN_DISPS(this->gameState.gfxCtx);
+    OPEN_DISPS(this->state.gfxCtx);
 
     gSPSegment(POLY_OPA_DISP++, 0x01, this->staticSegment);
 
-    Title_UpdateCounters(this);
-    Title_Draw(&this->gameState);
+    ConsoleLogo_UpdateCounters(this);
+    ConsoleLogo_Draw(&this->state);
     if (this->exit) {
         gSaveContext.seqIndex = (u8)NA_BGM_DISABLED;
         gSaveContext.nightSeqIndex = 0xFF;
         gSaveContext.gameMode = 1;
 
-        {
-            GameState* gameState = &this->gameState;
-            gameState->running = false;
-        }
-        SET_NEXT_GAMESTATE(&this->gameState, Opening_Init, OpeningContext);
+        STOP_GAMESTATE(&this->state);
+        NEXT_GAMESTATE(&this->state, TitleSetup_Init, sizeof(TitleSetupState));
     }
 
-    CLOSE_DISPS(this->gameState.gfxCtx);
+    CLOSE_DISPS(this->state.gfxCtx);
 }
 
-void Title_Destroy(GameState* thisx) {
-    TitleContext* this = (TitleContext*)thisx;
+void ConsoleLogo_Destroy(GameState* thisx) {
+    ConsoleLogoState* this = (ConsoleLogoState*)thisx;
 
-    Sram_InitSram(&this->gameState, &this->sramCtx);
+    Sram_InitSram(&this->state, &this->sramCtx);
     ShrinkWindow_Destroy();
     CIC6105_Nop80081828();
 }
 
-void Title_Init(GameState* thisx) {
-    TitleContext* this = (TitleContext*)thisx;
-    uintptr_t segmentSize =
-        (uintptr_t)_nintendo_rogo_staticSegmentRomEnd - (uintptr_t)_nintendo_rogo_staticSegmentRomStart;
+void ConsoleLogo_Init(GameState* thisx) {
+    ConsoleLogoState* this = (ConsoleLogoState*)thisx;
+    uintptr_t segmentSize = SEGMENT_ROM_SIZE(nintendo_rogo_static);
 
-    this->staticSegment = THA_AllocEndAlign16(&this->gameState.heap, segmentSize);
-    DmaMgr_SendRequest0(this->staticSegment, (uintptr_t)_nintendo_rogo_staticSegmentRomStart, segmentSize);
+    this->staticSegment = THA_AllocEndAlign16(&this->state.heap, segmentSize);
+    DmaMgr_SendRequest0(this->staticSegment, (uintptr_t)SEGMENT_ROM_START(nintendo_rogo_static), segmentSize);
```

Comment:
Is the cast needed? I suspect it is since we had to drop it from the macro ages ago.